pilot/cmd/pilot-agent/main.go

```diff
@@ -158,6 +158,7 @@ var (
 				return err
 			}
 			wait()
+			<-ctx.Done()
```

Comment:
This seems like we will hang forever if `wait()` terminates, and it looks like wait can terminate without ctx.Done

Reply:
good catch, i will fix